src/sentry/tasks/reports.py

```diff
@@ -804,10 +804,15 @@ def prepare_organization_report(timestamp, duration, organization_id, dry_run=Fa
         user_id__isnull=False, user__is_active=True
     ).exclude(flags=F("flags").bitor(OrganizationMember.flags["member-limit:restricted"]))
 
-    for user_id in member_set.values_list("user_id", flat=True):
+    if user_id:
```

Comment:
This is dangerous if the user id is not in the organization.

Reply:
I think something like this should be safer:
```py
kwargs = dict(user_id__isnull=False, user__is_active=True)
if user_id:
    kwargs["user_id"] = user_id

# If an OrganizationMember row doesn't have an associated user, this is
# actually a pending invitation, so no report should be delivered.
member_set = organization.member_set.filter(**kwargs).exclude(
    flags=F("flags").bitor(OrganizationMember.flags["member-limit:restricted"])
)

for user_id in member_set.values_list("user_id", flat=True):
    deliver_organization_user_report.delay(
        timestamp, duration, organization_id, user_id, dry_run=dry_run
    )
```

Reply:
Well it's not exactly dangerous, we have checks later down the stream: https://github.com/getsentry/sentry/blob/22796f6f112ee76e54d814fa650941f7fad66e30/src/sentry/tasks/reports.py#L950

If the user is not a part of the organization, projects will come out as empty and this check would fail, and nothing would happen.
But sure, this would work too.